### DISCUSSION TITLE
Does Copilot require that your code be hosted on Github?

### DISCUSSION THREAD

**Discussion by cesarbiods** [community/community#50830] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

My company is looking to test drive copilot but this led me to an important question that I haven't been able to find an answer for, does Copilot require that your code be hosted on Github? Some teams host their code on GitHub, but my team hosts our code on ADO. Does that mean I couldn't use copilot in my jet brains IDE? Thanks in advance!

**Comment by Akash1134** [comment, marked as answer]
Hi @cesarbiods ,

Copilot for Business does not retain any Code Snippets Data. GitHub Copilot transmits snippets of your code from your IDE to GitHub to provide Suggestions to you. Code snippets data is only transmitted in real-time to return Suggestions, and is discarded once a Suggestion is returned. 

User engagement data (which includes pseudonymous identifiers and general usage data), is required for the use of GitHub Copilot and will continue to be collected, processed, and shared with Microsoft and OpenAI when you use GitHub Copilot.

For more details, please read the [FAQs](https://github.com/features/copilot#:~:text=Privacy%20%E2%80%93%20Copilot%20for%20Business) and Copilot for Business [privacy statement](https://docs.github.com/en/site-policy/privacy-policies/github-copilot-for-business-privacy-statement).  

Thanks!

**Reply by jpwalton** [reply]
I think you misunderstood the question. They are trying to determine whether copilot requires that your _repositories_ are on github. I'm curious about the same thing

**Reply by cesarbiods** [reply]
You are correct in your assumption. That was my original question which was maybe but not fully answered by the reply above. > I think you misunderstood the question. They are trying to determine whether copilot requires that your _repositories_ are on github. I'm curious about the same thing

**Reply by Akash1134** [reply]
Copilot is trained on the code snippets that are on GitHub. Users have a [wide range of controls](https://docs.github.com/en/copilot/getting-started-with-github-copilot?tool=visualstudio) to _allow/ not alow_  Copilot to use their data. For more details, see our [privacy statement](https://docs.github.com/en/site-policy/privacy-policies/github-copilot-for-business-privacy-statement). Thanks!

**Reply by matt-buckley** [reply]
@Akash1134 to confirm as it is still not quite clear from the responses above - can you still use GitHub Copilot if your _repository_ is hosted somewhere other than GitHub? It's not a data privacy question but one of the prerequisites for using GitHub Copilot.

**Reply by kuroodo** [reply]
As a recent copilot user, I can confirm that you do not need to have your repository hosted on GitHub for copilot to work. All you need is a compatible IDE that supports the extension, and an internet connection. Should that ever change, I suggest using Codeium as an alternative since the base plan is free, SOC2 compliant, and on par with copilot.

**Reply by datanerd** [reply]
Just confirming that Copilot (for Individual or for Business) does not require your code to be hosted on GitHub. Copilot is an extension for your [supported IDE](https://docs.github.com/en/copilot/using-github-copilot/getting-started-with-github-copilot) so as long as a file can be opened in the IDE - you can use Copilot. The file can be in a repository on GitHub or any other code collaboration platform, or even just a file on your hard drive and not in a repository at all.

Upcoming Copilot for Enterprise has some features that will require access to the GitHub code and pull requests and as result will require your code to be in GitHub.

**Reply by filipeoliveira2w-alt** [reply]
and if it's inside an organization?
